processor/logdeduplicationprocessor/counter.go

```diff
@@ -0,0 +1,165 @@
+
+	return &logAggregator{
+		resources:         make(map[[16]byte]*resourceAggregator),
+		logCountAttribute: logCountAttribute,
+		timezone:          timezone,
+	}
+}
+
+// Export exports the counter as a Logs
+func (l *logAggregator) Export() plog.Logs {
+	logs := plog.NewLogs()
+
+	for _, resource := range l.resources {
+		resourceLogs := logs.ResourceLogs().AppendEmpty()
+		resourceAttrs := resourceLogs.Resource().Attributes()
+		resourceAttrs.EnsureCapacity(resourceAttrs.Len())
+		resource.attributes.CopyTo(resourceAttrs)
+
+		scopeLogs := resourceLogs.ScopeLogs().AppendEmpty()
+		for _, lc := range resource.logCounters {
+			lr := scopeLogs.LogRecords().AppendEmpty()
+
+			baseRecord := lc.logRecord
+
+			// Copy contents of base record
+			baseRecord.SetObservedTimestamp(pcommon.NewTimestampFromTime(lc.firstObservedTimestamp))
+			baseRecord.Body().CopyTo(lr.Body())
+
+			lr.Attributes().EnsureCapacity(baseRecord.Attributes().Len())
+			baseRecord.Attributes().CopyTo(lr.Attributes())
+
+			lr.SetSeverityNumber(baseRecord.SeverityNumber())
+			lr.SetSeverityText(baseRecord.SeverityText())
+
+			// Add attributes for log count and timestamps
+			lr.SetObservedTimestamp(pcommon.NewTimestampFromTime(lc.firstObservedTimestamp))
+			lr.SetTimestamp(pcommon.NewTimestampFromTime(timeNow()))
+			lr.Attributes().PutInt(l.logCountAttribute, lc.count)
+
+			firstTimestampStr := lc.firstObservedTimestamp.In(l.timezone).Format(time.RFC3339)
+			lastTimestampStr := lc.lastObservedTimestamp.In(l.timezone).Format(time.RFC3339)
+			lr.Attributes().PutStr(firstObservedTSAttr, firstTimestampStr)
+			lr.Attributes().PutStr(lastObservedTSAttr, lastTimestampStr)
+		}
+	}
+
+	return logs
+}
+
+// Add adds the logRecord to the resource aggregator that is identified by the resource attributes
+func (l *logAggregator) Add(resourceAttrs pcommon.Map, logRecord plog.LogRecord) {
+	key := pdatautil.MapHash(resourceAttrs)
+	resourceCounter, ok := l.resources[key]
+	if !ok {
+		resourceCounter = newResourceAggregator(resourceAttrs)
+		l.resources[key] = resourceCounter
+	}
+
+	resourceCounter.Add(logRecord)
+}
+
+// Reset resets the counter.
+func (l *logAggregator) Reset() {
+	l.resources = make(map[[16]byte]*resourceAggregator)
+}
+
+// resourceAggregator dimensions the counter by resource.
+type resourceAggregator struct {
+	attributes  pcommon.Map
+	logCounters map[[8]byte]*logCounter
+}
+
+// newResourceAggregator creates a new ResourceCounter.
+func newResourceAggregator(attributes pcommon.Map) *resourceAggregator {
+	return &resourceAggregator{
+		attributes:  attributes,
+		logCounters: make(map[[8]byte]*logCounter),
+	}
+}
+
+// Add increments the counter that the logRecord matches.
+func (r *resourceAggregator) Add(logRecord plog.LogRecord) {
+	key := getLogKey(logRecord)
+	lc, ok := r.logCounters[key]
+	if !ok {
+		lc = newLogCounter(logRecord)
+		lc.firstObservedTimestamp = timeNow().UTC()
+		r.logCounters[key] = lc
```

Comment:
nit: If `newLogCounter` handles the timestamp
```suggestion
		r.logCounters[key] = newLogCounter(logRecord)
```